examples/routing/src/Main.scala

```diff
@@ -0,0 +1,127 @@
+package myorg
+
+import cats.effect.IO
+import org.scalajs.dom
+import tyrian.Html.{param => _, _}
+import tyrian.*
+import urldsl.language.simpleErrorImpl.*
+import urldsl.vocabulary.Param
+import urldsl.vocabulary.Segment
+import urldsl.vocabulary.UrlMatching
+
+import scala.scalajs.js.annotation.JSExportTopLevel
+
+@JSExportTopLevel("TyrianApp")
+object HelloTyrian extends TyrianApp[Msg, Model]:
+
+  val homePath        = root / "home"
+  val counterPath     = root / "counter"
+  val moreComplexPath = root / "id" / segment[Int]
+  val pathWithParam   = (root / "user" / endOfSegments) ? param[Int]("age").?
+
+  private def getPageFromURL(path: String): Page =
+    homePath
+      .matchRawUrl(path)
+      .fold(
+        _ =>
+          counterPath
+            .matchRawUrl(path)
+            .fold(
+              _ =>
+                moreComplexPath
+                  .matchRawUrl(path)
+                  .fold(
+                    _ =>
+                      pathWithParam
+                        .matchRawUrl(path)
+                        .fold(
+                          _ => Page.NotFound,
+                          { case UrlMatching(_, ageOption) =>
+                            Page.UserAgePage(ageOption)
+                          }
+                        ),
+                    userId => Page.UserPage(userId)
+                  ),
+              _ => Page.Counter
+            ),
+        _ => Page.Home
+      )
+
+  def init(flags: Map[String, String]): (Model, Cmd[IO, Msg]) =
+    (
+      AppState(getPageFromURL(dom.window.location.href), counter = 0),
```

Comment:
My preferred pattern here would be to run this side effect in a `Cmd` on the line below, and that would instantly trigger and update.

Reply:
Thanks for your replies! I'm totally new into Elm architecture and don't know best patterns yet. I've updated sbt file. When I posted PR I've wrongly thought that the examples are "ad hoc" meaning that they don't belong to any sbt project. That was because I've only looked at `build.sbt` at root directory and didn't found `examples` reference. I realized that `examples` is `sbt` project on its own after your comment :) So sorry for the confusion.